apps/src/templates/projects/ReportAbusePopUp.jsx

```diff
@@ -0,0 +1,271 @@
+        {name: i18n.abuseTypeOther(), checked: false},
+      ],
+    });
+    this.cleanUpCaptcha();
+  }
+
+  onCaptchaVerification(token) {
+    this.setState({
+      submitButtonEnabled: true,
+      captchaCompleted: true,
+    });
+    this.token = token;
+  }
+
+  handleSubmit() {
+    this.sendReportAbuse();
+  }
+
+  sendReportAbuse() {
+    const formData = new FormData();
+
+    // name, email, age set in controller through current_user
+    formData.append(
+      'authenticity_token',
+      RailsAuthenticityToken.getRailsCSRFMetaTags().token
+    );
+    formData.append('channel_id', this.props.projectData.channel);
+    formData.append('abuse_url', this.props.abuseUrl);
+    formData.append('g-recaptcha-response', this.token);
+
+    const checkedCheckboxNames = this.state.checkboxes
+      .filter(checkbox => checkbox.checked)
+      .map(checkbox => checkbox.name);
+    formData.append('abuse_type', checkedCheckboxNames);
+
+    fetch('/report_abuse_pop_up', {
+      method: 'POST',
+      body: formData,
+    })
+      .then(response => {
+        if (response.ok) {
+          this.setState({
+            showReportConfirmation: true,
+          });
+          this.props.onReport();
+        } else if (response.status === 403) {
+          this.onCaptchaExpiration();
+        }
+      })
+      .catch(error => {
+        this.setState({
+          submitButtonEnabled: false,
+        });
+        console.error('Error sending report data:', error);
+      });
+  }
+
+  onCaptchaExpiration() {
+    this.setState({
+      submitButtonEnabled: false,
+      captchaCompleted: false,
+    });
+  }
+
+  handleCheckboxChange = checkboxName => {
+    this.setState(prevState => {
+      const updatedCheckboxes = prevState.checkboxes.map(checkbox =>
+        checkbox.name === checkboxName
+          ? {...checkbox, checked: !checkbox.checked}
+          : checkbox
+      );
+
+      const showRecaptcha = updatedCheckboxes.some(
+        checkbox => checkbox.checked
+      );
+
+      // if (!showRecaptcha) {
+      //   this.cleanUpCaptcha();
+      // }
+
+      return {
+        checkboxes: updatedCheckboxes,
+        showRecaptcha,
+      };
+    });
+  };
+
+  createCaptchaScript() {
+    const script = document.createElement('script');
+    script.src = 'https://www.google.com/recaptcha/api.js';
+    script.id = 'captcha';
+    script.async = true;
+    script.defer = true;
+    window.onCaptchaSubmit = token => this.onCaptchaVerification(token);
+    window.onCaptchaExpired = () => this.onCaptchaExpiration();
+    script.onload = () => this.setState({loadedCaptcha: true});
+    document.body.appendChild(script);
+    console.log('script added!');
+  }
+
+  render() {
+    const {
+      showRecaptcha,
+      checkboxes,
+      submitButtonEnabled,
+      showReportConfirmation,
+      loadedCaptcha,
+    } = this.state;
+
+    const captchaSiteKey = this.props.recaptchaSiteKey;
+
+    console.log('starting: ' + loadedCaptcha);
+
```

Comment:
I think you can remove lines 194-196 now that you added this to `componentDidUpdate`.

Reply:
Sorry about the back and forth on this - I'm also learning about the relationship between `render` and `componentDidUpdate`!